azure-mgmt-eventhub/tests/test_azure_mgmt_eventhub.py

```diff
@@ -0,0 +1,166 @@
+﻿# coding: utf-8
+
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+import unittest
+import time
+
+import azure.mgmt.eventhub.models
+from azure.mgmt.eventhub.models import EHNamespace
+from azure.mgmt.eventhub.models import EHNamespace, Sku, SkuName, AuthorizationRule, AccessRights, AccessKeys, Eventhub, CaptureDescription, Destination, EncodingCaptureDescription,  ErrorResponseException, ErrorResponse
+from azure.common.credentials import ServicePrincipalCredentials
+
+from devtools_testutils import AzureMgmtTestCase, ResourceGroupPreparer
+
+
+class MgmtEventHubTest(AzureMgmtTestCase):
+
+    def setUp(self):
+        super(MgmtEventHubTest, self).setUp()
+
+        self.eventhub_client = self.create_mgmt_client(
+            azure.mgmt.eventhub.EventHubManagementClient
+        )
+
+    @ResourceGroupPreparer()
+    def test_eh_eventhub_curd(self, resource_group, location):
+        # List all topic types
+        resource_group_name = resource_group.name #"ardsouza-resourcemovetest-group2"
+
+        # Create a Namespace
+        namespace_name = "testingpythontestcaseeventhubnamespaceEventhub"
+
+        namespaceparameter=EHNamespace(location,{'tag1':'value1','tag2':'value2'},Sku(SkuName.standard))
+        creatednamespace = self.eventhub_client.namespaces.create_or_update(resource_group_name, namespace_name, namespaceparameter, None, True).output
+        self.assertEqual(creatednamespace.name, namespace_name)
+
+        while (self.eventhub_client.namespaces.get(resource_group_name,namespace_name).provisioning_state != 'Succeeded'):
+            time.sleep(15)
+            continue
+
+        #
+        # # Get created Namespace
+        #
+        getnamespaceresponse = self.eventhub_client.namespaces.get(resource_group_name, namespace_name)
+        self.assertEqual(getnamespaceresponse.name, namespace_name)
+
+        # Create a Eventhub
+        eventhub_name = "testingpythontestcaseeventhub"
+        eventhubparameter = Eventhub(
+            message_retention_in_days=4,
+            partition_count=4,
+            capture_description=CaptureDescription(
+                enabled=True,
+                encoding=EncodingCaptureDescription.avro,
+                interval_in_seconds=120,
+                size_limit_in_bytes=10485763,
+                destination=Destination(
+                    name="EventHubArchive.AzureBlockBlob",
+                    storage_account_resource_id="/subscriptions/e2f361f0-3b27-4503-a9cc-21cfba380093/resourceGroups/Default-Storage-SouthCentralUS/providers/Microsoft.ClassicStorage/storageAccounts/arjunteststorage",
```

Comment:
You leak a subid here (security problem)
use self.settings.SUBSCRIPTION_ID while building the string

Reply:
Resolved